src/transforms/tag_cardinality_limit.rs

```diff
@@ -1,13 +1,8 @@
-use std::{
-    borrow::{Borrow, Cow},
-    collections::{HashMap, HashSet},
-    fmt,
-    future::ready,
-    pin::Pin,
-};
+use std::{collections::HashSet, fmt, future::ready, pin::Pin};
 
 use bloom::{BloomFilter, ASMS};
 use futures::{Stream, StreamExt};
+use hashbrown::HashMap;
```

Comment:
I don't think we need to use this crate explicitly. From the [docs](https://github.com/rust-lang/hashbrown#hashbrown):

> Since Rust 1.36, this is now the HashMap implementation for the Rust standard library. However you may still want to use this crate instead since it works in environments without std, such as embedded systems and kernels.

Reply:
This was used explicitly because the standard `HashMap` does not expose the same [`.entry_ref` functionality](https://docs.rs/hashbrown/latest/hashbrown/struct.HashMap.html#method.entry_ref) where we can get a reference to an entry without unconditionally copying the key (see `try_accept_tag`).

Reply:
Ah, that makes sense. Interesting!